core/network/fabric-interop-cc/contracts/interop/manage_assets.go

```diff
@@ -30,15 +32,57 @@ type AssetLockValue struct {
 	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
 }
 
-// function to generate contract id as "SHA256" hash based on the asset lock key (which is combination of asset type and asset it)
-func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+// Object used in the map, contractId --> <asset-type, num-units, locker, ...> (for fungible assets)
+type FungibleAssetLockValue struct {
+	Type		string	`json:"type"`
+	NumUnits	uint64	`json:"numUnits"`
+	Locker		string	`json:"locker"`
+	Recipient	string	`json:"recipient"`
+	Hash		string	`json:"hash"`
+	ExpiryTimeSecs	uint64	`json:"expiryTimeSecs"`
+}
+
+const(
+	assetKeyPrefix		= "AssetKey_"	// prefix for the map, asset-key --> asset-object
+	assetKeyDelimiter	= "_"		// delimiter for the asset-key
+	contractIdPrefix	= "ContractId_"	// prefix for the map, contractId --> asset-key
+)
+
+// function to generate a "SHA256" hash in base64 format for a given the preimage
+func generateSHA256HashInBase64Form(preimage string) string {
 	hasher := sha256.New()
-	assetLockKey := assetAgreement.Type + assetAgreement.Id
-	hasher.Write([]byte(assetLockKey))
+	hasher.Write([]byte(preimage))
 	shaHash := hasher.Sum(nil)
-	shaBase64 := base64.StdEncoding.EncodeToString(shaHash)
-	return assetLockKey, shaBase64
+	shaHashBase64 := base64.StdEncoding.EncodeToString(shaHash)
+	return shaHashBase64
 }
+
+/*
+ * Function to generate asset-lock key (which is combination of asset-type and asset-id)
+ * and contract-id (which is a hash on asset-lock key) for the non-fungible asset locking on the ledger
+ */
+func generateAssetLockKeyAndContractId(assetAgreement *common.AssetExchangeAgreement) (string, string) {
+	assetLockKey := assetKeyPrefix + assetAgreement.Type + assetKeyDelimiter + assetAgreement.Id
+	contractId := generateSHA256HashInBase64Form(assetLockKey)
+	return assetLockKey, contractId
+}
+
+/*
+ * Function to generate contract-id for fungible asset-locking on the ledger (which is
+ * a hash on the attributes of the fungible asset exchange agreement)
+ */
+func generateFungibleAssetLockContractId(assetAgreement *common.FungibleAssetExchangeAgreement) string {
+	// There can be multiple fungible asset lock agreement between the same parties for the same <asset-type, quantity>.
+	// Use the current time to generate a different hash if the request gets repeated. It's the caller's responsibility
+	// to ensure that the request didn't get repeated uniterntionally.
+	currentTimeSecs := uint64(time.Now().Unix())
+
+	preimage := assetAgreement.Type + strconv.Itoa(int(assetAgreement.NumUnits)) +
+			assetAgreement.Locker + assetAgreement.Recipient +  strconv.Itoa(int(currentTimeSecs))
```

Comment:
This won't work, because this combination of parameter values can be repeated in multiple lock instances (e.g., Alice locks 20 ERC tokens for Bob in multiple instances). So it will end up producing the same hash in multiple instances.

Reply:
@VRamakrishna, The hash generated also depends on `currentTimeSecs`. Hence even if the `FungibleAssetExchangeAgreement` parameters are same across different requests, we get different hash values. Made no changes.

Reply:
@VRamakrishna, replaced `currentTimeSecs` with the transactionId(https://pkg.go.dev/github.com/hyperledger/fabric-chaincode-go/shim#ChaincodeStub.GetTxID) to ensure uniqueness of the key across duplicate asset exchange agreements, and to maintain consistency of the write-set among multiple endorsements (`currentTimeSecs` might be different on different endorsing peers hence the write-sets from different endorsing peers may not match for the key `contractId` which is generated using `currentTimeSecs`).